_posts/2021-01-22-x-ways-to-write-meaningful-commit-messages.md

```diff
@@ -12,257 +12,215 @@ image:
+The reason behind using present tense is that the commit message is answering the question "What will happen after the commit is applied?".
+If we think of a commit as an independent patch, it doesn't matter if it applied in the past. What matters is that this patch is always supposed to make that particular change when it's applied.
 
 ### Detailed Enough
 Super-detailed commit messages are frustrating as well. We can find that level of detail in the code. For example, if our version control is Git, we can see all the changed files in Git, so we don't have to list them. 
 
 So, instead of answering "what are the changes?", **it's better to answer "What are the changes for?"**.
 
 ### Formatting
-Let's start with Git conventions. The rest of the styles have these conventions in their core. 
+Let's start with Git conventions. Other conventions usually have the Git conventions in their core. 
 
 **Git suggests a commit message should have three parts including a subject, a description, and a ticket number.** 
 Let's see the exact template mentioned in [Git's website](https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration):
 
-> Subject line (try to keep under 50 characters)
+```
+Subject line (try to keep under 50 characters)
 
-> Multi-line description of commit,
-> feel free to be detailed. (Up to 72)
- 
-> [Ticket: X]
+Multi-line description of commit,
+feel free to be detailed. (Up to 72)
 
+[Ticket: X]
+```
 
-**The subject is better to be less than 50 characters** to get a clean output when executing the following command: git log --oneline. **The description is better to be up to 72 characters.**
+**The subject is better to be less than 50 characters** to get a clean output when executing the command `git log --oneline`. **The description is better to be up to 72 characters.**
 
 Preslav Rachev [in his article](https://p5v.medium.com/what-s-with-the-50-72-rule-8a906f61f09c) explains the reason for the 50/72 rule. The ideal size of a git commit summary is around 50 characters in length. Analyzing the average length of commit messages in the Linux kernel suggests this number. The 72 character rule is to center the description on an 80-column terminal in the git log since it adds four blank spaces at the left when displaying the commit message, so we want to add space for four more blank spaces on the right side.
 
+## Conventional Commit Messages
 
-#### Conventional Commit Messages in Depth
-Firstly we discuss conventional commit messages in general, following by a brief review of what Semantic Versioning is. Then we explain some of the most famous styles.
+Let's now have a look at [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/), a specification that gives opinionated guar rails to format commit messages.
```

Comment:
I just realized this . "opinionated guar rails", I think should be "opinionated guardrails". Please correct me if I'm wrong.

Reply:
fixed it. If you find errors like these, you can just push fixes to the branch that's the base of this PR